jxl/src/render/stages/noise.rs

```diff
@@ -0,0 +1,293 @@
+// Copyright (c) the JPEG XL Project Authors. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// TODO(zond): remove once we use this!
+#![allow(dead_code)]
+
+use crate::{
+    features::noise::Noise,
+    render::{RenderPipelineInOutStage, RenderPipelineInPlaceStage, RenderPipelineStage},
+};
+
+pub struct ConvolveNoiseStage {
+    channel: usize,
+}
+
+impl ConvolveNoiseStage {
+    pub fn new(channel: usize) -> ConvolveNoiseStage {
+        ConvolveNoiseStage { channel }
+    }
+}
+
+impl std::fmt::Display for ConvolveNoiseStage {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "convolve noise for channel {}", self.channel,)
+    }
+}
+
+impl RenderPipelineStage for ConvolveNoiseStage {
+    type Type = RenderPipelineInOutStage<f32, f32, 2, 2, 0, 0>;
+
+    fn uses_channel(&self, c: usize) -> bool {
+        c == self.channel
+    }
+
+    // TODO: This
+    fn process_row_chunk(
+        &mut self,
+        _position: (usize, usize),
+        xsize: usize,
+        row: &mut [(&[&[f32]], &mut [&mut [f32]])],
+    ) {
+        for (input, output) in row {
```

Comment:
could be a `let (input, output) = row[0];` now

Reply:
Done